### PR TITLE
Offline behaviour kickstart

### PR DESCRIPTION
**Pull request by germanferrero**
This PR adds a wrapper component with children as a [render prop](https://reactjs.org/docs/render-props.html).
The wrapper calls props.children with isOnline parameter, consumed via API.
Also, as a first use case, an offline behaviour for registration component is included.

### CODE REVIEW

**Comment by gmarcos87** [comment]
Great, use #25 to get realtime status

File: web-app/src/components/utils/IsOnlineWrapper.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    componentDidMount() {
+        this.updateStatus();
+        this.interval = setInterval(() => this.updateStatus(), 1000);
+    }
+
+    render() {
+        return (
+            <>
+                {this.props.children(this.state.isOnline)}
```
Comment (gmarcos87) [inline web-app/src/components/utils/IsOnlineWrapper.js]: Why did you choose this option and not React.cloneElement ?

File: web-app/src/components/Registration/Registration.js [hunk 2]
```diff
@@ -84,14 +96,23 @@ class RegistrationPage extends React.Component {
...
     }
+
     render() {
         const { defaultNode, nodes } = this.state
         return (
-            <Registration handleSubmit={this.handleSubmit} defaultNode={defaultNode} nodes={nodes} handleGoToLogin={this.goToLogin} />
+            <IsOnlineWrapper>
+                {isOnline =>
+                    <>
+                        {isOnline}
```
Comment (gmarcos87) [inline web-app/src/components/Registration/Registration.js]: this is for debug?

Comment (germanferrero) [reply web-app/src/components/Registration/Registration.js]: oh yes, thanks!

File: web-app/src/components/utils/IsOnlineWrapper.js (hunk 1 above)
Comment (gmarcos87) [reply web-app/src/components/utils/IsOnlineWrapper.js]: I found this example using context 
https://gist.github.com/jvandenaardweg/58ed91e3c33de0b75a15d38853b23d7d

Comment (germanferrero) [reply web-app/src/components/utils/IsOnlineWrapper.js]: With React.cloneElement, I found that the syntax for using it wasn't explicit.
I got something like
```
<Wrapper>
    <Wrapped prop1={val1} ... propn={valn}> 
</Wrapper> 
```
And then taking this.props.isOnline at Wrapped.
Wasn't clear where it came from..

Context api looks cool. Anyway when needing to use more than one context, the syntax is the same as render props: https://reactjs.org/docs/context.html#consuming-multiple-contexts

I'm okay if we go with context api.

Comment (gmarcos87) [reply web-app/src/components/utils/IsOnlineWrapper.js]: I think a single context containing all the configuration and status of the app would be enough.

Comment (germanferrero) [reply web-app/src/components/utils/IsOnlineWrapper.js]: Ok, later on we can make the context general, please check if the new implementation looks good.

**Pull request merged**